### PR TITLE
Admin class: add tests

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Testing Instructions

---

See #328.

### CODE REVIEW

File: tests/phpunit/tests/Admin.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	 * @var int
+	 */
+	protected static $story_id;
+
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$admin_id      = $factory->user->create(
+			[ 'role' => 'administrator' ]
+		);
+
+		self::$story_id      = $factory->post->create(
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
		self::$story_id = $factory->post->create(
```

File: tests/phpunit/tests/Admin.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	/**
+	 * Story ID.
+	 *
+	 * @var int
+	 */
+	protected static $story_id;
+
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$admin_id      = $factory->user->create(
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
		self::$admin_id = $factory->user->create(
```

**Review by swissspidy — APPROVED** [review]

**Comment by spacedmonkey** [comment]
Any idea why the Integration tests are not working here @swissspidy ?

File: tests/phpunit/tests/Admin.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Tests;
+
+class Admin extends \WP_UnitTestCase {
+
+
+	/**
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
	/**
```

Comment (spacedmonkey) [reply tests/phpunit/tests/Admin.php]: I dont see what the change is here?

File: tests/phpunit/tests/Admin.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+	public function test_admin_body_class() {
+		$admin = new \Google\Web_Stories\Admin();
+		$admin->init();
+
+		wp_set_current_user( self::$admin_id );
+		set_current_screen( 'post.php' );
+		get_current_screen()->post_type = \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG;
+		get_current_screen()->base      = 'post';
+		$result                         = $admin->admin_body_class( 'current' );
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]: A bit simpler:

```suggestion
		$GLOBALS['current_screen'] = convert_to_screen( \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG );

		$result = $admin->admin_body_class( 'current' );
```

File: tests/phpunit/tests/Admin.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+		$_GET['from-web-story'] = self::$story_id;
+		$result                 = $admin->prefill_post_content( 'current' );
+		$poster                 = (string) wp_get_attachment_image_url( (int) get_post_thumbnail_id( self::$story_id ), \Google\Web_Stories\Media::STORY_POSTER_IMAGE_SIZE );
+		$this->assertContains( 'wp-block-web-stories-embed', $result );
+		$this->assertContains( $poster, $result );
+	}
+
+	public function test_prefill_post_content_invalid_user() {
+		$admin = new \Google\Web_Stories\Admin();
+		$admin->init();
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]: Calling `init` in all the tests here isn't really needed as you call the methods directly.
It's only needed in the `test_init()` test.

File: tests/phpunit/tests/Admin.php (hunk 4 above)
Comment (spacedmonkey) [reply tests/phpunit/tests/Admin.php]: I was looking for something like this!

File: tests/phpunit/tests/Admin.php (hunk 5 above)
Comment (swissspidy) [reply tests/phpunit/tests/Admin.php]:
```suggestion
```

File: tests/phpunit/tests/Admin.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+		wp_set_current_user( self::$admin_id );
+		$_GET['from-web-story'] = 999999999;
+		$result                 = $admin->prefill_post_content( 'current' );
+		$this->assertSame( 'current', $result );
+	}
+
+	public function test_prefill_post_title() {
+		$admin = new \Google\Web_Stories\Admin();
+		$admin->init();
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
```

File: tests/phpunit/tests/Admin.php [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+		wp_set_current_user( 0 );
+		$_GET['from-web-story'] = self::$story_id;
+		$result                 = $admin->prefill_post_content( 'current' );
+		$this->assertSame( 'current', $result );
+	}
+
+	public function test_prefill_post_content_invalid_id() {
+		$admin = new \Google\Web_Stories\Admin();
+		$admin->init();
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
```

File: tests/phpunit/tests/Admin.php [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+		set_current_screen( 'post.php' );
+		get_current_screen()->post_type = \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG;
+		get_current_screen()->base      = 'post';
+		$result                         = $admin->admin_body_class( 'current' );
+		$this->assertContains( 'folded', $result );
+	}
+
+	public function test_prefill_post_content() {
+		$admin = new \Google\Web_Stories\Admin();
+		$admin->init();
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
```

File: tests/phpunit/tests/Admin.php (hunk 3 above)
Comment (swissspidy) [reply tests/phpunit/tests/Admin.php]: Removing errant newlines between `class Admin extends \WP_UnitTestCase {` and the first method.

File: tests/phpunit/tests/Admin.php [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+	public function test_prefill_post_content_invalid_id() {
+		$admin = new \Google\Web_Stories\Admin();
+		wp_set_current_user( self::$admin_id );
+		$_GET['from-web-story'] = 999999999;
+		$result                 = $admin->prefill_post_content( 'current' );
+		$this->assertSame( 'current', $result );
+	}
+
+	/**
+	 * @covers ::prefill_post_content
```
Comment (swissspidy) [inline tests/phpunit/tests/Admin.php]:
```suggestion
	 * @covers ::prefill_post_title
```

**Pull request merged**